wsd/COOLWSD.cpp

```diff
@@ -924,7 +833,105 @@ class PrisonPoll : public TerminatingPoll
 
 /// This thread listens for and accepts prisoner kit processes.
 /// And also cleans up and balances the correct number of children.
-static std::unique_ptr<PrisonPoll> PrisonerPoll;
+static std::shared_ptr<PrisonPoll> PrisonerPoll;
+
+#if MOBILEAPP
+#ifndef IOS
+std::mutex COOLWSD::lokit_main_mutex;
+#endif
+#endif
+
+std::shared_ptr<ChildProcess> getNewChild_Blocks(SocketPoll &destPoll, unsigned mobileAppDocId)
+{
+    (void)mobileAppDocId;
+    const auto startTime = std::chrono::steady_clock::now();
+
+    std::unique_lock<std::mutex> lock(NewChildrenMutex);
+
+#if !MOBILEAPP
```

Comment:
I think it would be better if we split this function into mobile and non-mobile versions. The implementations are vastly different and it doesn't help us to have both together.